tests/test_wallet_address.py

```diff
@@ -691,239 +686,216 @@ def test_repr(wallet_address_factory):
     assert repr(wallet_address) == expected_repr
 
 
-@patch("cdp.Cdp.api_clients")
-@patch("cdp.Cdp.use_server_signer", True)
-def test_wallet_address_deploy_token_total_supply_string(mock_api_clients, wallet_address_factory):
+@patch("cdp.wallet_address.SmartContract")
+@patch("cdp.Cdp.use_server_signer", False)
+def test_wallet_address_deploy_token_total_supply_string(mock_smart_contract, wallet_address_factory):
     """Test the deploy_token method of a WalletAddress with a string total_supply."""
-    wallet_address = wallet_address_factory()
+    wallet_address = wallet_address_factory(key=True)
```

Comment:
btw i noticed some of these tests pass regardless of whether this is set @John-peterson-coinbase

Reply:
I think cause later the assertion that `sign` gets called just checks that wallet_address.key is what it was called with, but doesn't care whether it was set or not